pygis/docs/e_summarize_vector.md

```diff
@@ -0,0 +1,569 @@
+
+        # Check if index of point is in list of indices whose points have already been matched to a grid and counted
+        if i_2 in counted_points:
+
+            # If already counted, skip remaining statements in loop and start at top of loop
+            continue
+
+        # Otherwise, continue with remaining statements
+        else:
+            pass
+
+        # Get a well point by index
+        well = sc_county_wells.iloc[[i_2]]
+
+        # Reset index of well point (to 0) to match the index-reset cell
+        well = well.reset_index(drop = True)
+
+        # Check if well intersects the cell
+        # Best to use intersects instead of within or contains, as intersect will count points that fall exactly on the cell boundaries
+        # Points that fall exactly on a cell boundary (between two cells) will be allocated to the first of the two cells called in script
+        criteria_met = well.intersects(cell)[0]
+
+        # If preferred, can check if well is within cell or if cell contains well
+        # Both statements do the same thing
+        # criteria_met = well.within(cell)[0]
+        # criteria_met = cell.contains(well)[0]
+
+        # Check if criteria has been met (True)
+        if criteria_met:
+
+            # If True, increase counter by 1 for the cell
+            count += 1
+
+            # Add index of counted point to the list
+            counted_points.append(i_2)
+
+        # Otherwise, criteria is not met (False)
+        else:
+            pass
+
+    # Add total count for that cell to the list of counts
+    counts_list.append(count)
+
+# print(counts_list)
+
+# Add a new column to the grid GeoDataFrame with the list of counts for each cell
+sc_county_grid['Count'] = pd.Series(counts_list)
+
+# Display grid attribute table
+display(sc_county_grid)
+```
+
+We can check to make sure all well points in Santa Clara County were counted.
+
+```{code-cell} ipython3
+# Check total number of well points counted and compare to number of well points in input data
+print("Total number of well points counted: {}\nNumber of well points in input data: {}".format(sum(counts_list), len(sc_county_wells)))
+```
+
+Finally, we can plot the data.
+
+```{code-cell} ipython3
+# Create subplots
+fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+# Plot data
+sc_county.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+sc_county_wells.plot(ax = ax, marker = 'o', color = 'dimgray', markersize = 3)
+sc_county_grid.plot(ax = ax, column = "Count", cmap = "RdPu", edgecolor = 'lightseagreen', linewidth = 0.5, alpha = 0.70, legend = True)
+
+# Set title
+ax.set_title('Santa Clara County - Binning Well Points', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+## Kernel Density Estimation
+
+Kernel density estimation (KDE) uses a specified kernel function to visualize the density of points or polylines. For more information on KDE, check out [this visualization](https://mathisonian.github.io/kde/).
```

Comment:
this is really cool

Reply:
I do think we should go over the basics. Maybe go back to my GIS II slides or Hurleys?  There should be some graphics / text to pull here.

Reply:
At least need to cover what the main parameters are

Reply:
Sent some graphic ideas via email!